nav2_loopback_sim/nav2_loopback_sim/loopback_simulator.py

```diff
@@ -209,6 +230,143 @@ def debug(self, msg):
         self.get_logger().debug(msg)
         return
 
+    def get_map(self):
+        request = GetMap.Request()
+        if self.map_client.wait_for_service(timeout_sec=1.0):
+            # Request to get map
+            future = self.map_client.call_async(request)
+            rclpy.spin_until_future_complete(self, future)
+            if future.result() is not None:
+                self.map = future.result().map
+                self.get_logger().info(
+                    f'Received map data, {self.map.info.width}X{self.map.info.height}')
+            else:
+                self.get_logger().error('Failed to call map service')
+        else:
+            self.get_logger().error('Map service not available')
+
+    def get_laser_pose(self):
+        try:
+            if self.initial_pose is None:
+                return 0.0, 0.0, 0.0
+
+            if self.map is None:
+                self.get_map()
```

Comment:
Remove this, I'd try once on startup and after that just use max. This adds alot of latency